test/http-test.js

```diff
@@ -267,8 +267,108 @@ describe('HTTP', function() {
       isvalid: true,
       address: addr.toString(node.network),
       scriptPubKey: Script.fromAddress(addr).toRaw().toString('hex'),
-      ismine: false,
-      iswatchonly: false
+      iswitness: false,
+      isscript: false
+    });
+  });
+
+  it('should not validate invalid address', async () => {
+    // send an address for the wrong network
+    const json = await nclient.execute('validateaddress', [
+      addr.toString('main')
+    ]);
+    assert.deepStrictEqual(json, {
+      isvalid: false
+    });
+  });
+
+  it('should validate a p2wpkh address', async () => {
+    const info = await wallet.createAccount('foo3', {
+      witness: true
+    });
+    const json = await nclient.execute('validateaddress', [
+      info.receiveAddress
+    ]);
+    const addr = Address.fromString(info.receiveAddress);
+    const script = Script.fromAddress(addr);
+
+    assert.deepStrictEqual(json, {
+      isvalid: true,
+      iswitness: true,
+      address: info.receiveAddress,
+      isscript: addr.isScripthash(),
+      scriptPubKey: script.toJSON(),
+      witness_version: addr.version,
+      witness_program: addr.hash.toString('hex')
+    });
+  });
+
+  it('should validate a p2sh address', async () => {
+    await wallet.createAccount('foo4');
```

Comment:
Same as previously, I don't think it's  necessary to create account to generate public keys and addresses for the test, and the test will be more specific and less prone to unrelated errors from the wallet.

Reply:
👍